File: teste.py
import time

def sum_vec(arr):
    soma = 0
    for val in arr:
        soma += val
    return soma

tamanhos = [100, 1000, 10000]

print("--- Resultados em Python (perf_counter_ns - Nanossegundos) ---")

for n in tamanhos:
    vetor = [1] * n # Cria um vetor de tamanho 'n' preenchido com '1'
    
    print(f"\nTamanho do Vetor: {n}")
    
    for iteracao in range(1, 11):
        
        # Inicia o cronômetro de performance
        start = time.perf_counter_ns() 
        
        resultado = sum_vec(vetor) # Executa
        
        # Para o cronômetro
        end = time.perf_counter_ns() 
        
        tempo_ns = end - start
        
        print(f"Iteracao {iteracao:02d}: {tempo_ns} ns")

File: teste.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Função que soma o vetor
long long sum_vec(int *arr, int n) {
    long long soma = 0;
    for (int i = 0; i < n; i++) {
        soma += arr[i];
    }
    return soma;
}

int main() {
    int tamanhos[] = {100, 1000, 10000};
    int num_tamanhos = 3;
    struct timespec start, end;

    printf("--- Resultados em C (Nanossegundos) ---\n");

    for (int t = 0; t < num_tamanhos; t++) {
        int n = tamanhos[t];
        
        // Cria e preenche o vetor com o número 1
        int *vetor = (int *)malloc(n * sizeof(int));
        for (int i = 0; i < n; i++) vetor[i] = 1;

        printf("\nTamanho do Vetor: %d\n", n);
        
        for (int iteracao = 1; iteracao <= 10; iteracao++) {
            
            // Inicia o cronômetro de alta precisão
            clock_gettime(CLOCK_MONOTONIC, &start);
            
            long long resultado = sum_vec(vetor, n); // Executa
            
            // Para o cronômetro
            clock_gettime(CLOCK_MONOTONIC, &end);

            // Calcula a diferença exata em nanossegundos
            long long tempo_ns = (end.tv_sec - start.tv_sec) * 1000000000LL + (end.tv_nsec - start.tv_nsec);
            
            printf("Iteracao %02d: %lld ns\n", iteracao, tempo_ns);
        }
        free(vetor);
    }
    return 0;
}